calculator/src/Calculator.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { ERROR_MESSAGE, RULE } from './constants';
+
+class Calculator extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      num1: '',
+      num2: '',
+      operation: '',
+      total: localStorage.getItem('total') ?? 0,
+    };
+
+    window.addEventListener('beforeunload', this.handleBeforeUnload);
+    window.addEventListener('unload', this.handleUnload);
+  }
+
+  handleBeforeUnload = e => {
+    e.preventDefault();
+    e.returnValue = '';
+  };
+
+  handleUnload = () => {
+    if (!Number.isInteger(this.state.total)) return;
+
+    localStorage.setItem('total', this.state.total);
+  };
+
+  render() {
+    return (
+      <div className="calculator">
+        <h1 id="total">{this.state.total}</h1>
+        <div className="digits flex" onClick={this.handleDigitButtonClick}>
+          {this.digitsTemplate}
+        </div>
+        <div className="modifiers subgrid">
+          <button className="modifier" onClick={this.handleModifierButtonClick}>
+            AC
+          </button>
+        </div>
+        <div
+          className="operations subgrid"
+          onClick={this.handleOperationButtonClick}
+        >
+          {this.operationTemplate}
+        </div>
+      </div>
+    );
+  }
+
+  digitsTemplate = Array.from(Array(10).keys())
+    .reverse()
+    .map((digit, idx) => (
+      <button className="digit" key={idx}>
+        {digit}
+      </button>
+    ));
+
+  operationTemplate = ['/', 'X', '-', '+', '='].map((operation, idx) => (
+    <button className="operation" key={idx}>
+      {operation}
+    </button>
+  ));
+
+  handleDigitButtonClick = ({ target }) => {
+    const digit = target.textContent;
+    const { num1, num2, operation, total } = this.state;
+
+    if (!operation) {
+      if (num1.length >= RULE.MAX_DIGIT_LENGTH)
+        return alert(ERROR_MESSAGE.IS_OVER_MAX_DIGIT_LENGTH);
+
+      this.setState({
+        ...this.state,
+        num1: num1 + digit,
+        total: !Number(total) ? digit : total + digit,
+      });
+
+      return;
+    }
+
+    if (num2.length >= RULE.MAX_DIGIT_LENGTH)
+      return alert(ERROR_MESSAGE.IS_OVER_MAX_DIGIT_LENGTH);
+
+    this.setState({ ...this.state, num2: num2 + digit, total: total + digit });
+  };
+
+  handleModifierButtonClick = () => {
```

Comment:
ㅋㅋ저 궁금해서 AC 의 의미가 뭘까 찾아봤는데, All Clear 의 의미라고 하네요.
데모 없이 코드만 봤을 때 백스페이스같은 기능이 있나 했거든요. 아래처럼 네이밍 해보면 어떨까요?
```suggestion
  handleACButtonClick = () => {
  // handleAllClearButtonClick = () => {
```

Reply:
```js
<button className='modifier' onClick={this.handleModifierButtonClick}>
```

아하 모든 이벤트 핸들러명은 등록한 요소의 클래스명을 따랐습니다
(클래스 명은 사전에 주어졌어요 😊)

의미가 명확하게 드러나지 않는다면 변경하는 쪽이 더 좋겠네요~! b156fb20d134a987df374d2d8c820906fa632b33